File: python/tests/test_language.py
import pytest
from scanner.language import detect


def test_known_extensions():
    assert detect('file.py') == 'Python'
    assert detect('file.js') == 'JavaScript'
    assert detect('file.ts') == 'TypeScript'
    assert detect('file.go') == 'Go'
    assert detect('file.rs') == 'Rust'
    assert detect('file.rb') == 'Ruby'
    assert detect('file.java') == 'Java'
    assert detect('file.css') == 'CSS'
    assert detect('file.md') == 'Markdown'


def test_unknown_extension():
    assert detect('file.xyz') == 'unknown'
    assert detect('file.foobar') == 'unknown'
    assert detect('Makefile') == 'unknown'


def test_case_insensitive():
    assert detect('file.PY') == 'Python'
    assert detect('file.JS') == 'JavaScript'
    assert detect('file.TS') == 'TypeScript'


def test_dockerfile():
    assert detect('Dockerfile') == 'Dockerfile'
    assert detect('dockerfile') == 'Dockerfile'


def test_path_with_dirs():
    assert detect('/some/path/to/file.py') == 'Python'
    assert detect('src/components/App.tsx') == 'TSX'


def test_multiple_dots():
    # Only the last extension matters
    assert detect('file.test.py') == 'Python'
    assert detect('app.min.js') == 'JavaScript'

File: javascript/tests/test_language.js
import { test } from 'node:test';
import assert from 'node:assert/strict';
import { detectLanguage } from '../src/language.js';

test('known extensions', () => {
  assert.equal(detectLanguage('file.py'), 'Python');
  assert.equal(detectLanguage('file.js'), 'JavaScript');
  assert.equal(detectLanguage('file.ts'), 'TypeScript');
  assert.equal(detectLanguage('file.go'), 'Go');
  assert.equal(detectLanguage('file.rs'), 'Rust');
  assert.equal(detectLanguage('file.java'), 'Java');
  assert.equal(detectLanguage('file.css'), 'CSS');
  assert.equal(detectLanguage('file.md'), 'Markdown');
});

test('unknown extension', () => {
  assert.equal(detectLanguage('file.xyz'), 'unknown');
  assert.equal(detectLanguage('file.foobar'), 'unknown');
  assert.equal(detectLanguage('Makefile'), 'unknown');
});

test('case insensitive', () => {
  assert.equal(detectLanguage('file.PY'), 'Python');
  assert.equal(detectLanguage('file.JS'), 'JavaScript');
  assert.equal(detectLanguage('file.TS'), 'TypeScript');
});

test('Dockerfile detection', () => {
  assert.equal(detectLanguage('Dockerfile'), 'Dockerfile');
  assert.equal(detectLanguage('dockerfile'), 'Dockerfile');
});

test('path with directories', () => {
  assert.equal(detectLanguage('/some/path/file.py'), 'Python');
  assert.equal(detectLanguage('src/App.tsx'), 'TSX');
});

test('multiple dots — last extension wins', () => {
  assert.equal(detectLanguage('file.test.py'), 'Python');
  assert.equal(detectLanguage('app.min.js'), 'JavaScript');
});
